src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(object) {
+    if (
+      Object.getPrototypeOf(object) !== Carnivore
+      && object.hidden === false
+    ) {
+      object.health -= 50;
+    }
+
+    if (object.health <= 0) {
+      Animal.alive = Animal.alive.filter((element) => element.health > 0);
```

Comment:
This code would remove all animals with health <= 0 whenever any animal is bitten, not just the animal being bitten. Should filter out the specific animal when its health reaches 0.

consider using `Animal.alive.filter((element) => element !== object);